### DISCUSSION TITLE
Nested optimization of acquisition function

### DISCUSSION THREAD

**Discussion by HanyangHenry-Wang** [meta-pytorch/botorch#1646] [Q&A] [answered]
In Botorch, Knowledge Gradient is calculated by One-Shot method and I can understand the code.  But usually, KG is calculated by a nested optimization problem. My question is that now I would like to build a KG which uses nested optimization, is there any way to do so in Botorch? Or is there any acquisition function in Botorch that use nested optimization? Thank you.

**Comment by sdaulton** [comment, marked as answer]
I actually just implemented KG with nested optimization. I’ll put up a PR, hopefully in the next week

**Reply by HanyangHenry-Wang** [reply]
That will be very helpful, thank you!
